### PR TITLE
feat: show database options

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#3708 

## What's changed and what's your intention?

Main changes:

* Test creating tables with database `TTL`.
* Adds `options` column to `information_schema.schemata` table:
```sql
mysql> select * from information_schema.schemata;
+--------------+--------------------+----------------------------+------------------------+----------+--------------+
| catalog_name | schema_name        | default_character_set_name | default_collation_name | sql_path | options      |
+--------------+--------------------+----------------------------+------------------------+----------+--------------+
| greptime     | greptime_private   | utf8                       | utf8_bin               | NULL     |              |
| greptime     | hertzbeat          | utf8                       | utf8_bin               | NULL     | ttl='30days' |
| greptime     | information_schema | utf8                       | utf8_bin               | NULL     |              |
| greptime     | public             | utf8                       | utf8_bin               | NULL     |              |
+--------------+--------------------+----------------------------+------------------------+----------+--------------+
```
* Adds `SHOW FULL DATABASES` extension to show the database options:
```sql
 show full databases;
+--------------------+--------------+
| Database           | Options      |
+--------------------+--------------+
| greptime_private   |              |
| hertzbeat          | ttl='30days' |
| information_schema |              |
| public             |              |
+--------------------+--------------+
```

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

### CODE REVIEW

File: src/catalog/src/information_schema/schemata.rs [hunk 1]
```diff
@@ -160,21 +167,49 @@ impl InformationSchemaSchemataBuilder {
...
+                    .await
+                    .context(TableMetadataManagerSnafu)?;
+
+                let Some(schema_opts) = schema else {
+                    return SchemaNotFoundSnafu {
+                        catalog: &catalog_name,
+                        schema: &schema_name,
+                    }
+                    .fail();
+                };
```
Comment (fengjiachun) [inline src/catalog/src/information_schema/schemata.rs:187]:
```suggestion
                let schema_opts = table_metadata_manager
                    .schema_manager()
                    .get(SchemaNameKey::new(&catalog_name, &schema_name))
                    .await
                    .context(TableMetadataManagerSnafu)?
                    .context(SchemaNotFoundSnafu {
                        catalog: &catalog_name,
                        schema: &schema_name,
                    })?;
```

File: tests/cases/standalone/common/create/create_database_opts.result [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+|       |   TIME INDEX ("ts")                 |
+|       | )                                   |
+|       |                                     |
+|       | ENGINE=mito                         |
+|       | WITH(                               |
+|       |   ttl = '1h'                        |
+|       | )                                   |
++-------+-------------------------------------+
+
+USe public;
```
Comment (fengjiachun) [inline tests/cases/standalone/common/create/create_database_opts.result:53]: Is the use of `USe` intentional?

File: tests/cases/standalone/common/create/create_database_opts.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+SHOW FULL DATABASES;
+
+USE mydb;
+
+CREATE TABLE test(host STRING, cpu DOUBLE, ts TIMESTAMP TIME INDEX);
+
+SHOW CREATE TABLE test;
+
+USe public;
```
Comment (fengjiachun) [inline tests/cases/standalone/common/create/create_database_opts.sql:13]: Is the use of `USe` intentional?

**Review by fengjiachun — APPROVED** [review]

File: tests/cases/standalone/common/create/create_database_opts.result (hunk 2 above)
Comment (killme2008) [reply tests/cases/standalone/common/create/create_database_opts.result:53]: Good catch!

File: src/common/meta/src/key/schema_name.rs [hunk 4]
```diff
@@ -57,6 +57,17 @@ pub struct SchemaNameValue {
     pub ttl: Option<Duration>,
 }
 
+impl Display for SchemaNameValue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(ttl) = self.ttl {
+            let ttl = humantime::format_duration(ttl);
+            write!(f, "ttl='{ttl}'",)?;
```
Comment (WenyXu) [inline src/common/meta/src/key/schema_name.rs:64]:
```suggestion
            write!(f, "ttl='{ttl}'")?;
```

**Review by WenyXu — APPROVED** [review]

**Pull request merged**